lib/Service/MembersService.php

```diff
@@ -387,8 +389,18 @@ private function verifyIdentContact(&$ident, $type) {
 	 */
 	private function addGroupMembers(Circle $circle, $groupId) {
 
-		$group = OC::$server->getGroupManager()
-							->get($groupId);
+		$groupManager = OC::$server->getGroupManager();
+		$group = $groupManager->get($groupId);
+
+		$user = OC::$server->getUserSession()->getUser();
+
+		if (!$this->configService->isAddingAnyGroupMembersAllowed() &&
+			$group instanceof IGroup && $user instanceof IUser &&
+			!$group->inGroup($user) && !$groupManager->isAdmin($user->getUID())
+		) {
```

Comment:
I am not a fan of this condition; let's assume that user is not null, I prefer having an unhandled exception than a failed condition. something like if _($group !== null && [...]->isAddingAnyGroupMembersAllowed() && not in group && not admin)_